### PR TITLE
Github Workflow added

### PR DESCRIPTION
**Pull request by AtharvaThorve**
Resolves issue #13<br>
Runs tests on push to master and creation of new pull request to master.<br>
Not all tests present in `server/test` are able to pass.<br>
The workflow runs tests on `ubuntu-latest, macOS-latest, and windows-latest` on node version 8, 10 and 12<br>
Also, few edge cases in Events API are fixed.

### CODE REVIEW

**Comment by ShubhankarKG** [comment]
Could you list out some of the tests that failed so that I can look into those ? 
Thanks.

**Comment by AtharvaThorve** [comment]
The event's add form fails the test.

**Comment by AtharvaThorve** [comment]
Changed matrix os to ubuntu-latest only as GitHub action for MongoDB is supported only there.

File: server/src/controllers/Events.js [hunk 1]
```diff
@@ -54,7 +55,10 @@ module.exports = {
       const eventId = req.params.id;
       const file = req.file;
       if (file) {
-        await cloudinary.v2.uploader.destroy(eventId);
+        try {
+          await cloudinary.api.resource(eventId);
+          await cloudinary.v2.uploader.destroy(eventId);
+        } catch(error) {}
```
Comment (ShubhankarKG) [inline server/src/controllers/Events.js:61]: It's better if send an error response here, presumably with a 500 response since something failed on the server side.

File: server/src/controllers/Events.js [hunk 2]
```diff
@@ -65,10 +69,9 @@ module.exports = {
           public_id: image.public_id
         };
       }
-      const event = await Event.findByIdAndUpdate(eventId, req.body);
-      res.json({
-        id: event._id
-      });
+      let event = await Event.updateOne({_id: mongoose.Types.ObjectId(eventId)}, req.body);
+      event = await Event.findById(eventId);
```
Comment (ShubhankarKG) [inline server/src/controllers/Events.js:73]: Any special reasons for changing the update function here ? Both look the same to me tbh.

File: server/src/controllers/Events.js [hunk 3]
```diff
@@ -79,8 +82,11 @@ module.exports = {
     const eventId = req.params.id;
     const event = await Event.findById(eventId);
     await event.remove();
-    await cloudinary.v2.uploader.destroy(eventId);
-    res.status(204);
+    try {
+      await cloudinary.api.resource(eventId);
+      await cloudinary.v2.uploader.destroy(eventId);
+    } catch(error) {}
```
Comment (ShubhankarKG) [inline server/src/controllers/Events.js:88]: It's better if send an error response here, presumably with a 500 response since something failed on the server side.

File: server/src/controllers/Events.js (hunk 2 above)
Comment (AtharvaThorve) [reply server/src/controllers/Events.js:73]: It was not giving proper output so had to break it into two different steps.

File: server/src/controllers/Events.js [hunk 4]
```diff
@@ -79,8 +86,15 @@ module.exports = {
...
-    await cloudinary.v2.uploader.destroy(eventId);
-    res.status(204);
+    try {
+      await cloudinary.api.resource(eventId);
+      try {
+        await cloudinary.v2.uploader.destroy(eventId);
+      } catch(error) {
+        res.status(500).json({});
+      }
+    } catch(error) {}
```
Comment (AtharvaThorve) [inline server/src/controllers/Events.js:96]: This change is added as the first try-catch block checks whether the image exists or not and the nested try-catch block checks if a server-side error occurred during the deletion of image.

**Comment by AtharvaThorve** [comment]
Also, there is no use for the add form API in our app so maybe we should remove it.

**Comment by ShubhankarKG** [comment]
I presume the add form api was added to store the links of google forms that we create. Will have to look into whether we really need it or not.

**Comment by AtharvaThorve** [comment]
@ShubhankarKG Thanks a lot for the review.
